src/app/password/new-password-and-confirm/new-password-and-confirm.component.spec.ts

```diff
@@ -1,119 +1,139 @@
     fixture.detectChanges();
+    newPasswordAndConfirmHarness = await TestbedHarnessEnvironment.harnessForFixture(
+      fixture,
+      NewPasswordAndConfirmHarness
+    );
   });
-  passwordValidation();
+  newPasswordValidation();
   confirmPasswordValidation();
 });
 
-function passwordValidation() {
-  it('should show password required error', () => {
-    setPasswordAndExpectError('', 'required');
-  });
-
-  it('should show password length error', () => {
-    setPasswordAndExpectError('1234567', 'minlength');
-  });
-
-  it('should show password pattern error', () => {
-    setPasswordAndExpectError('1234567', 'pattern');
-  });
+function newPasswordValidation() {
+  passwordIsMissing();
+  passwordIsValid();
+  passwordErrorCases();
+}
 
-  it('should not show any error when password is valid', () => {
-    setNewPasswordValue('Abcd1234');
-    expect(getNewPasswordFormControl().errors).toBeNull();
+function passwordIsMissing(): void {
+  describe('password is missing', () => {
+    it('should show password required error', async () => {
+      await newPasswordAndConfirmHarness.setNewPassword('');
+      expect(await newPasswordAndConfirmHarness.isNewPasswordRequiredErrorDisplayed()).toBeTrue();
+    });
   });
 }
 
-function setPasswordAndExpectError(password: string, errorName: string): void {
-  setNewPasswordValue(password);
-  expect(newPasswordHasError(errorName)).toBeTrue();
+function passwordIsValid(): void {
+  describe('password is valid', () => {
+    it('should not show any error', async () => {
+      await newPasswordAndConfirmHarness.setNewPassword(VALID_PASSWORD);
+      expect(await newPasswordAndConfirmHarness.isNewPasswordRequiredErrorDisplayed()).toBeFalse();
+    });
+  });
 }
 
-function setNewPasswordValue(value: string): void {
-  setFormControlValue(
-    component,
-    NewPasswordAndConfirmComponent.NEW_PASSWORD_FORM_CONTROL_NAME,
-    value
-  );
+function passwordErrorCases(): void {
+  const errorCases = [
+    {
+      descriptionText: 'missing a letter',
+      password: INVALID_PASSWORD_MISSING_LETTER,
+      expectedErrors: new PasswordErrors(true, false, false, false)
+    },
+    {
+      descriptionText: 'missing a number',
+      password: INVALID_PASSWORD_MISSING_NUMBER,
+      expectedErrors: new PasswordErrors(false, true, false, false)
+    },
+    {
+      descriptionText: 'missing a symbol',
+      password: INVALID_PASSWORD_MISSING_SYMBOL,
+      expectedErrors: new PasswordErrors(false, false, true, false)
+    },
+    {
+      descriptionText: 'too short',
+      password: INVALID_PASSWORD_TOO_SHORT,
+      expectedErrors: new PasswordErrors(false, false, false, true)
+    }
+  ];
+  errorCases.forEach(({ descriptionText, password, expectedErrors }) => {
+    describe(`password is ${descriptionText}`, () => {
+      it(`should indicate password is ${descriptionText}`, async () => {
+        await setPasswordAndExpectErrors(password, expectedErrors);
+      });
+    });
+  });
```

Comment:
Change block to follow Arrange-Act-Assert pattern (https://github.com/CareMessagePlatform/jasmine-styleguide#arrange-act-assert)
Add beforeEach() to call the action (```await newPasswordAndConfirmHarness.setNewPassword(password);```), and then check the state inside the it() (```await checkPasswordRequirements(passwordErrors);```).